xclim/core/calendar.py

```diff
@@ -106,23 +108,27 @@ def convert_calendar(
     Parameters
     ----------
     source : xr.DataArray
-      Input array/dataset with a time coordinate of a valid dtype (datetime64 or a cftime.datetime)
+      Input array/dataset with a time coordinate of a valid dtype (datetime64 or a cftime.datetime).
     target : Union[xr.DataArray, str]
       Either a calendar name or the 1D time coordinate to convert to.
       If an array is provided, the output will be reindexed using it and in that case, days in `target`
-      that are missing in the converted `source` are filled by NaNs.
+      that are missing in the converted `source` are filled by `missing` (which defaults to NaN).
     align_on : {None, 'date', 'year'}
       Must be specified when either source or target is a `360_day` calendar, ignored otherwise. See Notes.
+    missing : Optional[any]
+      A value to use for filling in dates in the target that were missing in the source.
+      If `target` is a string, default (None) is not to fill values. If it is an array, default is to fill with NaN.
```

Comment:
```suggestion
      A value to use for filling in dates in the target that are missing in the source.
      If `target` is a string, default (None) is to not fill any values. 
      If `target` is an array, default is to fill with numpy.NaN/pandas.na.
```
Not all `NaN`s are made the same. Numpy's `NaN` is float-based, but in the event of an integer variable, does numpy's `np,nan` know to keep the nantype (integer) synonymous with the rest of the values in the array?

Reply:
True, numpy and xarray do not know how to have a nan type similar to the values. However, using np.nan is xarray's default already. It will recast an entire int array to float. But using `pandas.NA` would anyway recast the int array to an `object` one, which is not better!
Here, passing `target : DataArray, missing = np.nan` has the exact same behavior as not passing `missing` (the previous behavior).
Until xarray/numpy do something about their non-float NaNs, I suggest we stick to what they do.

Reply:
Okay. It's definitely something to take note of. I can see this being a bit jarring in the case of users converting their calendar and seeing the size of the files jump when introducing NaNs. Perhaps an `Note` is a good idea in the docstring ?